src/test/fuzz/prevector.cpp

```diff
@@ -0,0 +1,268 @@
+    }
+
+    Size size() const {
+        return real_vector.size();
+    }
+
+    Size capacity() const {
+        return pre_vector.capacity();
+    }
+
+    void shrink_to_fit() {
+        pre_vector.shrink_to_fit();
+    }
+
+    void swap() {
+        real_vector.swap(real_vector_alt);
+        pre_vector.swap(pre_vector_alt);
+    }
+
+    void move() {
+        real_vector = std::move(real_vector_alt);
+        real_vector_alt.clear();
+        pre_vector = std::move(pre_vector_alt);
+        pre_vector_alt.clear();
+    }
+
+    void copy() {
+        real_vector = real_vector_alt;
+        pre_vector = pre_vector_alt;
+    }
+
+    void resize_uninitialized(realtype values) {
+        size_t r = values.size();
+        size_t s = real_vector.size() / 2;
+        if (real_vector.capacity() < s + r) {
+            real_vector.reserve(s + r);
+        }
+        real_vector.resize(s);
+        pre_vector.resize_uninitialized(s);
+        for (auto v : values) {
+            real_vector.push_back(v);
+        }
+        auto p = pre_vector.size();
+        pre_vector.resize_uninitialized(p + r);
+        for (auto v : values) {
+            pre_vector[p] = v;
+            ++p;
+        }
+    }
+};
+
+}
+
+void test_one_input(const std::vector<uint8_t>& buffer)
+{
+    FuzzedDataProvider prov(buffer.data(), buffer.size());
+    prevector_tester<8, int> test;
+
+    while (prov.remaining_bytes()) {
+        switch (prov.ConsumeIntegralInRange<int>(0, 15 + 3 * (test.size() > 0))) {
+        case 0:
+            test.insert(prov.ConsumeIntegralInRange<size_t>(0, test.size()), prov.ConsumeIntegral<int>());
+            break;
+        case 16:
+            test.erase(prov.ConsumeIntegralInRange<size_t>(0, test.size() - 1));
+            break;
+        case 1:
+            test.resize(std::max(0, std::min(30, (int)test.size() + prov.ConsumeIntegralInRange<int>(0, 4) - 2)));
+            break;
+        case 2:
+            test.insert(prov.ConsumeIntegralInRange<size_t>(0, test.size()), 1 + prov.ConsumeBool(), prov.ConsumeIntegral<int>());
+            break;
+        case 3: {
+            int del = std::min<int>(test.size(), 1 + prov.ConsumeBool());
+            int beg = prov.ConsumeIntegralInRange<int>(0, test.size() - del);
+            test.erase(beg, beg + del);
+            break;
+        }
+        case 4:
+            test.push_back(prov.ConsumeIntegral<int>());
+            break;
+        case 17:
+            test.pop_back();
+            break;
+        case 5: {
+            int values[4];
+            int num = 1 + prov.ConsumeIntegralInRange<int>(0, 3);
+            for (int k = 0; k < num; ++k) {
+                values[k] = prov.ConsumeIntegral<int>();
+            }
+            test.insert_range(prov.ConsumeIntegralInRange<size_t>(0, test.size()), values, values + num);
+            break;
+        }
+        case 6: {
+            int del = std::min<int>(test.size(), 1 + prov.ConsumeIntegralInRange<int>(0, 3));
```

Comment:
Could this be combined with `case 3` and generalized?
```suggestion
            int del = prov.ConsumeIntegralInRange<int>(1, test.size());
```

Reply:
Indeed, done (and the del range can start at 0 even).